beacon_node/lighthouse_network/src/rpc/handler.rs

```diff
@@ -855,6 +855,45 @@ where
         }
 
         let (req, substream) = substream;
+        let current_fork = self.fork_context.current_fork();
+        let spec = &self.fork_context.spec;
+
+        match &req {
+            RequestType::BlocksByRange(request) => {
+                let max_allowed = spec.max_request_blocks(current_fork) as u64;
```

Comment:
Have we ever thought about making the limit dependent on the protocol version or the active fork at the time of the `request.slot` range? It seems like we could inadvertently reject requests from clients that use different limits for pre-Deneb blocks.

Reply:
Hi Michael, can you elaborate on what you mean by `request.slot` range? 
If it's pre Deneb `max_request_blocks` will return the value for [pre Deneb](https://github.com/sigp/lighthouse/blob/c228e57caaab5d63aeacf89cda5cf64ebe6975ce/consensus/types/src/chain_spec.rs#L624) right?

Reply:
We are always using the current fork according to wall clock time, so if the request is for pre-Deneb data but the current fork is post-Deneb, we will use the Deneb value.

Reply:
I think we don't do it that way to keep things consistent between byroot and byrange requests (since we cannot know the slot when we get root requests).
I asked other CL teams how they handle it and it seems like prysm and lodestar do by slot, but nimbus does it by current fork.
In practice, I don't think this matters that much because all clients request 1/2 epochs per by_range request which doesn't hit the limits.